### PR TITLE
Implement provenance methods

### PR DESCRIPTION
**Pull request by eruizgar91**
## Description

Add provenance methods.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [x] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

File: tests/metadata/test_metadata.py [hunk 1]
```diff
@@ -107,21 +107,21 @@ def test_text_search_invalid_query():
...
-                   num_matches + 1)
-
-    metadata_provider.publish_asset_ddo(asset2)
-
-    assert len(metadata_provider.query_search(search_query={"query": {"type": ["access"]}},
-                                              offset=10000)['results']) == (
-                   num_matches + 2)
-    metadata_provider.retire_asset_ddo(asset1.did)
-    metadata_provider.retire_asset_ddo(asset2.did)
+# def test_query_search(asset1, asset2):
```
Comment (r-marques) [inline tests/metadata/test_metadata.py:110]: Should this be removed if it's not being used?

File: tests/nevermined/test_provenance.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
+import logging
+import uuid
+
+
+def new_did():
+    return uuid.uuid4().hex + uuid.uuid4().hex
+
+
+def test_get_provenance(publisher_instance, ddo_sample):
+    logging.debug("".format())
```
Comment (r-marques) [inline tests/nevermined/test_provenance.py:10]: I don't think this is doing anything, right?

**Review by r-marques — APPROVED** [review]
lgtm

File: tests/nevermined/test_provenance.py (hunk 2 above)
Comment (eruizgar91) [reply tests/nevermined/test_provenance.py:10]: Yes, I think that I got it from other test. I will get rid off

File: tests/metadata/test_metadata.py (hunk 1 above)
Comment (eruizgar91) [reply tests/metadata/test_metadata.py:110]: Sure

**Pull request merged**